src/compiler/crystal/syntax/parser.cr

```diff
@@ -4147,6 +4147,7 @@ module Crystal
         block_arg = call_args.block_arg
         named_args = call_args.named_args
         has_parentheses = call_args.has_parentheses
+        force_call ||= has_parentheses || (args.try(&.empty?) == false) || (named_args.try(&.empty?) == false)
```

Comment:
What happens if this is always set to `true` here?

Reply:
Yeah, that could actually work.

Reply:
It breaks for `x = 1; foo x do\nend`: `x` is considered a call.

This happens because `call_args` is still present even if every property is nil or false except `stopped_on_do_after_space`.

Reply:
Makes sense, thanks!